File: 백준/Silver/1931. 회의실 배정/회의실 배정.swift
let n = Int(readLine()!)!

var arr:[[Int]] = Array(repeating: Array(repeating: 0, count: 2), count: n)

for i in 0...n-1{
    let input = readLine()!.split(separator: " ").map{Int(String($0))!}
    arr[i][0] = input[0]
    arr[i][1] = input[1]
}

arr.sort(by: {
    if($0[1] == $1[1]){
        return $0[0] < $1[0]
    }
    return $0[1] < $1[1]
})

var cnt = 1
var end = arr[0][1]

for i in 1...arr.count-1{
    
    var start = arr[i][0]
    if(start >= end){
        end = arr[i][1]
        cnt += 1
    }
}
print(cnt)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.io.*;
import java.util.*;

public class Main {


    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());
        int cnt = 0;
        int [][] arr = new int[N][2];
        for (int i = 0; i < N; i++) {
            String [] A = br.readLine().split(" ");
            arr[i][0] = Integer.parseInt(A[0]);
            arr[i][1] = Integer.parseInt(A[1]);
        }

        Arrays.sort(arr, (o1,o2) ->{
            if(o1[1] == o2[1]){
                return o1[0] - o2[0];
            }
            else {
                return o1[1] - o2[1];
            }
        });

        int index = 0;

        for (int i = 0; i < N; i++) {
            if(i == 0){
                index = arr[i][1];
                cnt++;
            }
            else {
                if(index <= arr[i][0]){
                    index = arr[i][1];
                    cnt++;
                }
            }
        }

        System.out.println(cnt);


    }




}
